cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/InMemoryStreamMetaStore.java

```diff
@@ -51,10 +55,18 @@ public boolean streamExists(String accountId, String streamName) throws Exceptio
   }
 
   @Override
-  public synchronized List<StreamSpecification> listStreams() throws Exception {
-    ImmutableList.Builder<StreamSpecification> builder = ImmutableList.builder();
-    for (String stream : streams.values()) {
-      builder.add(new StreamSpecification.Builder().setName(stream).create());
+  public synchronized Multimap<NamespaceMeta, StreamSpecification> listStreams() throws Exception {
```

Comment:
Would it be better to make this ``Multimap<String, StreamSpecification>``? I don't think we will use fields other than ``id`` here, but maybe it is better to keep it as is to guard for a future case when we would need other fields?